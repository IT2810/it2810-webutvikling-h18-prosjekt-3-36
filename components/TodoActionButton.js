import React from "react";
import { Icon } from "expo";
import { Platform } from "react-native";
import { FloatingAction } from "react-native-floating-action";
import Categories from '../constants/Categories';
import Colors from '../constants/Colors';

export default class TodoActionButton extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    
    // actions er hvilken handlinger du kan gjøre etter du har trykker på den store + knappen. Vi har valgt å ha kategorier her.
    // Så bruker vi konstanten vår Categories for å hente ut hvilket ikon og hvilken farge etc etc de skal ha.
    const actions = [
      {
        text: "Work",
        name: "work",
        color: "white",
        icon: 
        <Icon.Ionicons
          name={(Platform.OS === 'ios' ? "ios-" : "md-") + Categories["work"].icon}
          size={30}
          color={Categories["work"].color}
        />,
        position: 1
      },
      {
        text: "School",
        name: "school",
        color: "white",
        icon:
          <Icon.Ionicons
            name={(Platform.OS === 'ios' ? "ios-" : "md-") + Categories["school"].icon}
            size={30}
            color={Categories["school"].color}
          />,
        position: 2
      },
      {
        text: "Fun",
        name: "fun",
        color: "white",
        icon:
          <Icon.Ionicons
            name={(Platform.OS === 'ios' ? "ios-" : "md-") + Categories["fun"].icon}
            size={30}
            color={Categories["fun"].color}
          />,
        position: 3
      },
      {
        text: "Other",
        name: "other",
        color: "white",
        icon:
          <Icon.Ionicons
            name={(Platform.OS === 'ios' ? "ios-" : "md-") + Categories["other"].icon}
            size={30}
            color={Categories["other"].color}
          />,
        position: 4
      }
    ];
    return (
      <FloatingAction
        color={Colors.tintColor}
        actions={actions}
        position="right"
        onPressItem={name =>
          this.props.navigate("TodoFormModal", {
            category: name
          })
        }
      />
    );
  }
}
