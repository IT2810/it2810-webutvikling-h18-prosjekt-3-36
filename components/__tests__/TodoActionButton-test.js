import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import TodoActionButton from "../TodoActionButton";


it("TodoActionButton Matches snapshot.", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<TodoActionButton />)
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
});
