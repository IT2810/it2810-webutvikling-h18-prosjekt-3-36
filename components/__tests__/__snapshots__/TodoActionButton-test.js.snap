// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoActionButton Matches snapshot. 1`] = `
<AnimatedComponent
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "elevation": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 0,
      },
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
>
  <AnimatedComponent
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 85,
          "position": "absolute",
          "zIndex": 10,
        },
        Object {
          "alignItems": "flex-end",
          "right": -1000,
        },
        Object {
          "opacity": 0,
        },
        Object {
          "bottom": 94,
        },
      ]
    }
  >
    <FloatingActionItem
      active={false}
      color="red"
      distanceToEdge={30}
      margin={8}
      name="work"
      onPress={[Function]}
      paddingTopBottom={8}
      position="right"
      text="Work"
      textBackground="#ffffff"
      textColor="#444444"
      textElevation={5}
    />
    <FloatingActionItem
      active={false}
      color="blue"
      distanceToEdge={30}
      margin={8}
      name="school"
      onPress={[Function]}
      paddingTopBottom={8}
      position="right"
      text="School"
      textBackground="#ffffff"
      textColor="#444444"
      textElevation={5}
    />
    <FloatingActionItem
      active={false}
      color="orange"
      distanceToEdge={30}
      margin={8}
      name="fun"
      onPress={[Function]}
      paddingTopBottom={8}
      position="right"
      text="Fun"
      textBackground="#ffffff"
      textColor="#444444"
      textElevation={5}
    />
    <FloatingActionItem
      active={false}
      color="green"
      distanceToEdge={30}
      margin={8}
      name="other"
      onPress={[Function]}
      paddingTopBottom={8}
      position="right"
      text="Other"
      textBackground="#ffffff"
      textColor="#444444"
      textElevation={5}
    />
  </AnimatedComponent>
  <AnimatedComponent
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 28,
          "elevation": 5,
          "height": 56,
          "justifyContent": "center",
          "overflow": "visible",
          "position": "absolute",
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 5,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 3,
          "width": 56,
          "zIndex": 2,
        },
        Object {},
        Object {
          "backgroundColor": "#1253bc",
          "bottom": 30,
          "right": 30,
        },
        Object {
          "opacity": 1,
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
            Object {
              "scale": 1,
            },
          ],
        },
      ]
    }
  >
    <TouchableOpacity
      activeOpacity={0.85}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 28,
          "height": 56,
          "justifyContent": "center",
          "width": 56,
          "zIndex": 3,
        }
      }
    >
      <AnimatedComponent
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 28,
              "flex": 1,
              "height": 56,
              "justifyContent": "center",
              "width": 56,
            },
            Object {
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
            },
          ]
        }
      >
        <AddIcon
          height={15}
          width={15}
        />
      </AnimatedComponent>
    </TouchableOpacity>
  </AnimatedComponent>
</AnimatedComponent>
`;
